commit c4149b75f4646ec8a4ca190f9e22c2ef54066828
Author: John Levon <john.levon@joyent.com>
Date:   2019-05-07T19:51:53+00:00 (5 months ago)
    
    TRITON-1353 Triton systems should be able to disable SMT
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/docs/index.md b/docs/index.md
index 778fd27..5628c0b 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -124,6 +124,8 @@ specialized circumstances in production.
 | **ALLOC_WEIGHT_UNRESERVED_DISK**   | Float | 1.0   | Bias selection towards CNs with more unreserved disk.                        |
 | **ALLOC_WEIGHT_UNRESERVED_RAM**    | Float | 2.0   | Bias selection towards CNs with more unreserved memory.                      |
 | **FEATURE_USE_CNAGENT_COMMAND_EXECUTE** | Boolean | false | Experimental: Use cn-agent's command_execute function instead of Ur when available. |
+| **FEATURE_ENABLE_VIRTUAL_SERVERS_FILTER** | Boolean | false | Whether to enable the virtual server filtering. When enabled, during DAPI allocation, all virtual servers (e.g. mockcloud servers) will be filtered out if the vm is a docker vm, or the vm has the `triton.placement.exclude_virtual_servers` tag set to true. This virtual server filtering is needed for Triton testing, but should not be enabled for production environments. |
+| **SMT_ENABLED_DEFAULT**	| Boolean | true | The default simultaneous multi-threading mode for newly-installed CNs. |
 
 If any of the keys above aren't in the `sdc` `metadata` section, it's treated as
 if the default value was specified. Be careful when changing from the default
diff --git a/docs/static.md b/docs/static.md
index 91d2e56..439bd97 100644
--- a/docs/static.md
+++ b/docs/static.md
@@ -93,6 +93,8 @@ specialized circumstances in production.
 | **ALLOC_WEIGHT_UNRESERVED_DISK**   | Float | 1.0   | Bias selection towards CNs with more unreserved disk.                        |
 | **ALLOC_WEIGHT_UNRESERVED_RAM**    | Float | 2.0   | Bias selection towards CNs with more unreserved memory.                      |
 | **FEATURE_USE_CNAGENT_COMMAND_EXECUTE** | Boolean | false | Experimental: Use cn-agent's command_execute function instead of Ur when available. |
+| **FEATURE_ENABLE_VIRTUAL_SERVERS_FILTER** | Boolean | false | Whether to enable the virtual server filtering. When enabled, during DAPI allocation, all virtual servers (e.g. mockcloud servers) will be filtered out if the vm is a docker vm, or the vm has the `triton.placement.exclude_virtual_servers` tag set to true. This virtual server filtering is needed for Triton testing, but should not be enabled for production environments. |
+| **SMT_ENABLED_DEFAULT**	| Boolean | true | The default simultaneous multi-threading mode for newly-installed CNs. |
 
 If any of the keys above aren't in the `sdc` `metadata` section, it's treated as
 if the default value was specified. Be careful when changing from the default
diff --git a/lib/models/server.js b/lib/models/server.js
index 20d23c7..eef32ca 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -1259,6 +1259,18 @@ ModelServer.getBootParamsDefault = function (callback) {
                 params.kernel_flags[k] = server.kernel_flags[k];
             }
 
+            /*
+             * We want to default new / factory-reset CNs to picking up the SAPI
+             * default. The default boot params are originally generated via
+             * initializeBuckets(), but we want this policy to apply on a DC
+             * upgrade, not just a completely fresh install. So we directly mix
+             * in the SAPI default here.
+             */
+            if (!params.kernel_args.hasOwnProperty('smt_enabled')) {
+                params.kernel_args['smt_enabled'] =
+                    ModelServer.getConfig().cnapi.smt_enabled_default;
+            }
+
             params.boot_modules = server.boot_modules;
 
             params.default_console = server.default_console;
diff --git a/package.json b/package.json
index 9a4cd66..23045e3 100644
--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "async": "1.5.2",
     "backoff": "2.5.0",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#066e52618ed6f4ef25121716b95b19cb8f90a5b4",
+    "dapi": "git+https://github.com/joyent/sdc-designation.git#18117e9395bf7ae410d445bca1a7a32690d99320",
     "deep-object-diff": "1.1.0",
     "dox": "0.9.0",
     "gc-stats": "1.2.0",
diff --git a/sapi_manifests/cnapi/template b/sapi_manifests/cnapi/template
index cecd8cd..222823f 100644
--- a/sapi_manifests/cnapi/template
+++ b/sapi_manifests/cnapi/template
@@ -43,7 +43,8 @@
 		"url": "http://{{{assets_admin_ips}}}"
 	},
 	"cnapi": {
-		"url": "http://{{{auto.ADMIN_IP}}}"
+		"url": "http://{{{auto.ADMIN_IP}}}",
+                "smt_enabled_default": {{{SMT_ENABLED_DEFAULT}}}{{^SMT_ENABLED_DEFAULT}}true{{/SMT_ENABLED_DEFAULT}}
 	},
 	"imgapi": {
 		"url": "http://{{{IMGAPI_SERVICE}}}"
